var/spack/repos/builtin/packages/ipopt/package.py

```diff
@@ -113,7 +113,7 @@ def configure_args(self):
                     ]
                 )
 
-        if "coinhsl" in spec:
+        if spec.satisfies("+coinhsl"):
```

Comment:
This seems incorrect and maybe should be:
```suggestion
        if spec.satisfies("^coinhsl"):
```

Reply:
in this case, it's equivalent since it `depends_on(coninhsl)` when `+coinhsl`